test/lib/usd/translators/testUsdExportLight.py

```diff
@@ -0,0 +1,187 @@
+
+        self.assertTrue(Gf.IsClose(sphereLight.GetColorAttr().Get(1), Gf.Vec3f(0.3, 1, 0.2), 1e-6))
+        self.assertTrue(Gf.IsClose(sphereLight.GetColorAttr().Get(5), Gf.Vec3f(0, 0.2, 0.1), 1e-6))
+
+        self.assertTrue(Gf.IsClose(sphereLight.GetIntensityAttr().Get(1), 0.8, 1e-6))
+        self.assertTrue(Gf.IsClose(sphereLight.GetDiffuseAttr().Get(1), 0, 1e-6))
+        self.assertTrue(Gf.IsClose(sphereLight.GetTreatAsPointAttr().Get(1), 1, 1e-6))
+        self.assertTrue(Gf.IsClose(sphereLight.GetRadiusAttr().Get(1), 0, 1e-6))
+    
+        (translateOp,rotateOp) = sphereLight.GetOrderedXformOps()
+        self.assertEqual(translateOp.GetOpType(), UsdGeom.XformOp.TypeTranslate)
+        self.assertEqual(rotateOp.GetOpType(), UsdGeom.XformOp.TypeRotateXYZ)
+        self.assertTrue(UsdGeom.XformCommonAPI(sphereLight))
+        self.assertTrue(Gf.IsClose(translateOp.Get(1), Gf.Vec3d(10, 7, -8), 1e-6))
+        self.assertTrue(Gf.IsClose(translateOp.Get(5), Gf.Vec3d(5, 3, 8), 1e-6))
+        self.assertTrue(Gf.IsClose(rotateOp.Get(1), Gf.Vec3f(-45, 90, -5), 1e-6))
+        
+        self.assertTrue(lightPrim.HasAPI(UsdLux.ShadowAPI))
+        shadowAPI = UsdLux.ShadowAPI(lightPrim)
+        self.assertTrue(shadowAPI)
+
+        self.assertTrue(lightPrim.HasAPI(UsdLux.ShapingAPI))
+        shapingAPI = UsdLux.ShapingAPI(lightPrim)
+        self.assertTrue(shapingAPI)
+        self.assertTrue(Gf.IsClose(shapingAPI.GetShapingConeAngleAttr().Get(1), 25, 1e-6))
+        self.assertTrue(Gf.IsClose(shapingAPI.GetShapingConeSoftnessAttr().Get(1), 0.4, 1e-6))
+        self.assertTrue(Gf.IsClose(shapingAPI.GetShapingFocusAttr().Get(1), 8, 1e-6))
+
+    def _ValidateAreaLight(self):
+        lightPrimPath = '/areaLight1'
+        lightPrim = self._stage.GetPrimAtPath(lightPrimPath)
+        self.assertTrue(lightPrim)
+        rectLight = UsdLux.RectLight(lightPrim)
+        self.assertTrue(rectLight)
+        self.assertTrue(Gf.IsClose(rectLight.GetColorAttr().Get(), Gf.Vec3f(0.8, 0.7, 0.6), 1e-6))
+        self.assertTrue(Gf.IsClose(rectLight.GetIntensityAttr().Get(), 1.2, 1e-6))
+        self.assertTrue(rectLight.GetNormalizeAttr().Get())
+        
+        self.assertTrue(lightPrim.HasAPI(UsdLux.ShadowAPI))
+        shadowAPI = UsdLux.ShadowAPI(lightPrim)
+        self.assertTrue(shadowAPI)
+
+        (translateOp,rotateOp,scaleOp) = rectLight.GetOrderedXformOps()
+        self.assertEqual(translateOp.GetOpType(), UsdGeom.XformOp.TypeTranslate)
+        self.assertEqual(rotateOp.GetOpType(), UsdGeom.XformOp.TypeRotateXYZ)
+        self.assertEqual(scaleOp.GetOpType(), UsdGeom.XformOp.TypeScale)
+        self.assertTrue(UsdGeom.XformCommonAPI(rectLight))
+        self.assertTrue(Gf.IsClose(translateOp.Get(1), Gf.Vec3d(8, 0, 10), 1e-6))
+        self.assertTrue(Gf.IsClose(scaleOp.Get(1), Gf.Vec3f(4,3,2), 1e-6))
+        self.assertTrue(Gf.IsClose(rotateOp.Get(1), Gf.Vec3f(0,23,0), 1e-6))
+        
+    def testExportLights(self):
```

Comment:
How about adding a roundtrip test?

Reply:
It's already more or less what is being done, as the reader test uses the output file I got from the export test. Since I'm testing all the parameters, in theory anything that breaks should fail either the reader or the writer test, so I'm not sure an additional roundtrip test would actually test more that what is already done. But yes, we can add another test if desired

Reply:
I don't know how this applies to lights since they probably won't change that much...but with time USD files may get out of date with what exporter would generate and we may miss roundtripping issues.

Reply:
Added a roundtrip test in the last commit